### PR TITLE
Refine contribution.md to be clearly

### LINKED ISSUE #1888: 完善 CONTRIBUTION.md
### 你当前使用的版本

latest

### 描述一下此特性

最近我们收到一些非常棒的 PR，例如：

- https://github.com/halo-dev/halo/pull/1865
- https://github.com/halo-dev/halo/pull/1874
- https://github.com/halo-dev/halo/pull/1882
- https://github.com/halo-dev/halo/pull/1883
- https://github.com/halo-dev/halo/pull/1887

部分作者可能第一次参与开源，缺乏一些经验。与此同时，我们的 [CONTRIBUTION.md](https://github.com/halo-dev/halo/blob/1f75ba09f8c1f3ad6b3514299508bcd8ab1270d6/CONTRIBUTING.md) 也相对比较简陋。

因此，我们非常有必要丰富 CONTRIBUTION.md 文档来帮助 PR 作者更好地参与开源。

### 附加信息

_No response_

### PR DESCRIPTION
**Pull request by JohnNiang**
### What this PR does

As I mentioned in the title, I refactored documentation in CONTRIBUTION.md file to be clearer.

### Which issue does this PR fix

Fix #1888

/kind documentation
/cc @halo-dev/sig-halo

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,64 +1,108 @@
...
+   点击 Halo 仓库主页右上角的 `Fork` 按钮即可。
 
-```bash
-git push origin {BRANCH_NAME}
-```
+2. Clone 仓库到本地
 
-#### 5. 提交 pull request
+   ```bash
+   git clone https://github.com/{YOUR_USERNAME}/halo
```
Comment (ruibaby) [inline CONTRIBUTING.md:33]: 是否可以直接简写为：

```
git clone https://github.com/{YOUR_USERNAME}/halo --recursive
```

https://docs.halo.run/developer-guide/core/prepare#%E5%85%8B%E9%9A%86%E9%A1%B9%E7%9B%AEoptional

**Review by ruibaby — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (JohnNiang) [reply CONTRIBUTING.md]: 已优化，请看：<https://github.com/halo-dev/halo/pull/1890/commits/92b1ac9bf7283b94f2c71075e426b7e9d67ebfe9>。

**Review by ruibaby — COMMENTED** [review]
/lgtm

**Review by guqing — APPROVED** [review]
/approve

**Review by ruibaby — APPROVED** [review]

**Pull request merged**